charts/service-deployment/templates/deployment.yaml

```diff
@@ -80,12 +80,21 @@ spec:
         resources:
           {{- toYaml .Values.resources | nindent 10 }}
 
-        {{- if ne .Values.readinessProbe.httpGet.path "" }}
+        {{- if or (ne .Values.readinessProbe.httpGet.path "") (.Values.readinessProbe.exec.command) }}
         readinessProbe:
-          httpGet:
-            path: {{ .Values.readinessProbe.httpGet.path }}
-            port: {{ .Values.service.port }}
-            scheme: HTTP
+          {{- if ne .Values.readinessProbe.httpGet.path "" }}
+            httpGet:
+              path: {{ .Values.readinessProbe.httpGet.path }}
+              port: {{ .Values.service.port }}
+              scheme: HTTP
+          {{- end }}
+          {{- if .Values.readinessProbe.exec.command }}
```

Comment:
I guess this is just checking that the entry exists but what if the array is zero length?  Should this be a length > 0 sort of check given the input type?